acra/configuring-maintaining/installing/building-acrawriter.md

```diff
@@ -0,0 +1,262 @@
+
+### Building AcraWriter for Ruby
+
+> Before building AcraWriter for Ruby, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Installation
+
+```bash
+gem install acrawriter
+```    
+
+#### Test
+```bash
+gem install --user-install minitest rake
+rake test
+```
+Output
+```
+Finished in 0.003829s, 261.1501 runs/s, 0.0000 assertions/s.
+
+1 runs, 0 assertions, 0 failures, 0 errors, 0 skips
+```    
+
+#### AcraWriter supports ActiveDirectory
+
+```
+gem 'activerecord_acrawriter'
+```
+
+#### Examples
+[With Zones](https://github.com/cossacklabs/acra/blob/master/examples/ruby/example_with_zone.rb) and [without Zones](https://github.com/cossacklabs/acra/blob/master/examples/ruby/example.rb).
+
+A real-world example project that protects Ruby on Rails application is available in [Acra example projects](https://github.com/cossacklabs/acra-engineering-demo#example-4-protecting-data-in-a-rails-application).
+
+### Building AcraWriter for Nodejs
+
+> Before building AcraWriter for NodeJS, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Installation
+```bash
+sudo apt install node npm
+sudo apt install nodejs-legacy
+sudo npm install -g npm
+cd wrappers/nodejs
+npm install nan
+npm install
+```
+#### Testing
+```
+node test.js
+```
+Output
+```
+work
+```
+#### Examples
+[With Zones](https://github.com/cossacklabs/acra/blob/master/examples/nodejs/examples_with_zone.js) and [without Zones](https://github.com/cossacklabs/acra/blob/master/examples/nodejs/examples_without_zones.js).
+
+### Building AcraWriter for PHP
+
+> Before building AcraWriter for PHP, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Installation
+
+```bash
+sudo apt install php5 php5-dev
+cd themis
+sudo make phpthemis_install
+# extension=phpthemis.so
+sudo nano /etc/php5/cli/php.ini
+# check that extension is available
+php --ri phpthemis
+cd -
+```
+#### Testing
+```bash
+php acra/wrappers/php/test.php
+```
+Output
+```
+work
+```
+Now use `acra/wrappers/php/acrawriter.php` in your project. In future
+`acrawriter` will contain composer.json and will be added to packagist.
+
+#### Examples
+[With Zones](https://github.com/cossacklabs/acra/blob/master/examples/php/example_with_zone.php) and [without Zones](https://github.com/cossacklabs/acra/blob/master/examples/php/example.php)
+
+### Building AcraWriter for Go
+
+> Before building AcraWriter for Go, make sure you've installed [Themis as system library]({{< ref "themis/installation/" >}})!
+
+#### Installation
+```bash
+sudo apt install golang
+# leave your GOPATH or set some
+export GOPATH=$HOME/work
+go get github.com/cossacklabs/acra/acra-writer
+```   
```

Comment:
Same as elsewhere, I don't think users need to be taught here how to install Go and configure their GOPATH. Just `go get` should be enough (as a second item after a link to GoThemis installation instructions).